patroni/postgresql/__init__.py

```diff
@@ -1259,26 +1259,41 @@ def promote(self, wait_seconds: int, task: CriticalTask,
 
     @staticmethod
     def _wal_position(is_primary: bool, wal_position: int,
-                      received_location: Optional[int], replayed_location: Optional[int]) -> int:
-        return wal_position if is_primary else max(received_location or 0, replayed_location or 0)
+                      receive_lsn: Optional[int], replay_lsn: Optional[int]) -> int:
+        return wal_position if is_primary else max(receive_lsn or 0, replay_lsn or 0)
 
     def timeline_wal_position(self) -> Tuple[int, int, Optional[int]]:
         # This method could be called from different threads (simultaneously with some other `_query` calls).
         # If it is called not from main thread we will create a new cursor to execute statement.
         if current_thread().ident == self.__thread_ident:
             timeline = self._cluster_info_state_get('timeline') or 0
             wal_position = self._cluster_info_state_get('wal_position') or 0
-            replayed_location = self.replayed_location()
-            received_location = self.received_location()
+            replay_lsn = self.replay_lsn()
+            receive_lsn = self.receive_lsn()
             pg_control_timeline = self._cluster_info_state_get('pg_control_timeline')
         else:
-            timeline, wal_position, replayed_location, received_location, _, pg_control_timeline, _, write_location = \
+            timeline, wal_position, replay_lsn, receive_lsn, _, pg_control_timeline, _, write_location = \
                 self._query(self.cluster_info_query)[0][:8]
-            received_location = max(received_location or 0, write_location or 0)
+            receive_lsn = max(receive_lsn or 0, write_location or 0)
 
-        wal_position = self._wal_position(bool(timeline), wal_position, received_location, replayed_location)
+        wal_position = self._wal_position(bool(timeline), wal_position, receive_lsn, replay_lsn)
         return timeline, wal_position, pg_control_timeline
 
+    def replica_wal_positions(self) -> Tuple[Optional[int], Optional[int]]:
+        """Get receive and replay LSN.
+
+        :returns: a tuple composed of 2 integers representing receive and replay LSN (if available).
+        """
+        # This method could be called from different threads (simultaneously with some other `_query` calls).
+        # If it is called not from main thread we will create a new cursor to execute statement.
+        if current_thread().ident == self.__thread_ident:
+            replay_lsn = self.replay_lsn()
+            receive_lsn = self.receive_lsn()
+        else:
+            _, _, replay_lsn, receive_lsn = self._query(self.cluster_info_query)[0][:4]
+
+        return receive_lsn, replay_lsn
```

Comment:
I think we better not have this method, because when called not from the main thread, on standby, it will result in two separate queries :(
It would be better to return everything what is required from timeline_wal_position() method. It is used only in one place, in ha.py and it shouldn't create any problem.